### PR TITLE
feat: option to use ByteArray instead of Array<felt252>

### LINKED ISSUE #9: Option to use `ByteArray` instead of `Array<felt252>`
# Description
When we created this tool a while only available option to return strings was to return `Array<felt252>`.

# Additional Information
[ High Difficulty - Rust - Low Priority]

# TODO
Implement an option that will be used like this : 
```
cargo run -- generate path/to/yourfile.svg --type ByteArray
```
Option will default to **`Array<felt252>`**.

With `--type` option specified, program will then return generated cairo with type passed in option.

# Acceptance Criteria
- [ ] Add an option to change cairo base type (used in functions input and output)
- [ ] Program is usable without the option specified (defaulting to `Array<felt252>`)
- [ ] Program is usable with option specified and will fail if type other than `ByteArray` is specified
- [ ] Code decoupling to write cairo program should be decoupled for different type handlers
- [ ] Common code should be then generic

### PR DESCRIPTION
**Pull request by TropicalDog17**
### Description

Closes #9 

### Checklist

- [x]  Add an option to change cairo base type (used in functions input and output)
- [x]  Program is usable without the option specified (defaulting to Array<felt252>)
- [x]  Program is usable with option specified and will fail if type other than ByteArray is specified
- [ ]  Code decoupling to write cairo program should be decoupled for different type handlers
- [ ]  Common code should be then generic

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ pub struct SvgElement {
     pub attributes: HashMap<String, String>,
     pub replacement: HashMap<String, String>,
     pub nodes: Vec<SvgElement>,
+    pub type_override: Option<String>,
```
Comment (MartianGreed) [inline src/parser/mod.rs]: I think we should use an enum for that kind of handling. Enum can implement `Display` trait, it should simplify handling of logic afterwards

**Review by MartianGreed — COMMENTED** [review]
Great work, just one minor comment ! Keep rockin'

File: src/parser/mod.rs (hunk 1 above)
Comment (TropicalDog17) [reply src/parser/mod.rs]: I'm still quite new to Rust, thanks a lot for your advice!

File: src/writer/mod.rs [hunk 2]
```diff
@@ -4,6 +4,8 @@ use thiserror::Error;
 
 pub mod cairo;
 
+pub const DEFAULT_STRING_CAIRO_TYPE: &str = "Array<felt252>";
+
```
Comment (MartianGreed) [inline src/writer/mod.rs]: I guess this can be removed now !

File: test-2.cairo
Comment (MartianGreed) [inline test-2.cairo:1]: this file should be ignored !

File: test.cairo
Comment (MartianGreed) [inline test.cairo:1]: this file can be ignored too or placed into an `examples/cairo` directory

**Review by MartianGreed — APPROVED** [review]

File: examples/test-2.cairo [hunk 3]
```diff
@@ -0,0 +1,66 @@
+////////////////////////////////////////
+//            SVGenerated             //
+////////////////////////////////////////
+// source file: test-2.svg
+
+
+#[inline(always)]
+fn print_51729(ref svg: Array<felt252>, data: @Data) {
+	svg.append('<g><text><tspan>Carbon units</t'
+	svg.append('span></text></g>'
```
Comment (MartianGreed) [inline examples/test-2.cairo]: can we fix this file please ?

**Review by MartianGreed — CHANGES_REQUESTED** [review]

**Review by MartianGreed — APPROVED** [review]

**Pull request merged**